### PR TITLE
feat: add support for security context

### PR DESCRIPTION
**Pull request by mowies**
### This PR
- adds the (container) security context to the chart generation

### CODE REVIEW

**Comment by mowies** [comment]
@arttor this PR should be ready for review :)

File: examples/operator/templates/deployment.yaml [hunk 1]
```diff
@@ -91,7 +91,20 @@ spec:
         resources: {{- toYaml .Values.controllerManager.manager.resources | nindent 10
           }}
         securityContext:
-          allowPrivilegeEscalation: false
+          allowPrivilegeEscalation: {{ .Values.controllerManager.manager.containerSecurityContext.allowPrivilegeEscalation
```
Comment (arttor) [inline examples/operator/templates/deployment.yaml:94]: I think that we don't have to template each field individually here. It is more readable to render the whole yaml for securityContext from values.yaml. In the same way, as it is done above for resources. Or for [nodeSelector here](https://github.com/arttor/helmify/commit/db73ef44705129b0663316226ca860865cd3fa74)

**Review by arttor — CHANGES_REQUESTED** [review]

File: examples/operator/templates/deployment.yaml (hunk 1 above)
Comment (mowies) [reply examples/operator/templates/deployment.yaml]: done! :)

**Comment by arttor** [comment]
@mowies looks really nice and ready to merge. i am just thinking is there any case when chart maintainer doesn't want securityContext to be modified. In this case we should add flag and add securityContext to template only if it is enabled. what do you think?

**Comment by mowies** [comment]
I don't really see a use case for the CLI flag honestly.. I think if you have the security context set up, you will probably also want to show those values in the helm chart. Maybe we can leave that for now and see if users request it in the future?

**Review by arttor — DISMISSED** [review]

**Comment by arttor** [comment]
@mowies can you please update your branch? i cannot merge pr due to conflicts

**Comment by mowies** [comment]
@arttor done

**Comment by arttor** [comment]
@mowies still can't 
image

**Comment by mowies** [comment]
ok this is very weird. for me, it shows everything green with no conflicts. i will do a full rebase...

Screenshot 2023-03-15 at 10 04 43

**Comment by arttor** [comment]
strange, i still can't merge. can you try to rebase main?

**Comment by mowies** [comment]
ok, all done. seem that some examples were not fully re-generated with the last PR. I regenerated both the operator and the sample app again.

**Review by arttor — APPROVED** [review]

**Comment by arttor** [comment]
@mowies thank you! Your changes are available in [v0.3.29 release](https://github.com/arttor/helmify/releases/tag/v0.3.29)

**Pull request merged**